tools/nvqpp/nvq++.in

```diff
@@ -584,6 +607,16 @@ trap delete_temp_files EXIT
 
 COMPILER_FLAGS="${CPPSTD} -Wno-c++20-extensions ${COMPILER_FLAGS}"
 
+# If the user has requested NVQC, choose a reasonable default. Note that Nvidia
+# GPUs do not need to be installed in order to select an Nvidia-powered backend
+# when using NVQC.
+if ${NVQC}; then
+	NVQIR_SIMULATION_BACKEND="custatevec-fp32"
+	if [ -z "${TARGET_CONFIG}" ]; then
+		TARGET_CONFIG=nvidia
```

Comment:
Why is this set to nvidia here, just to be set to nvqc further below? It looks like the only consequence this has is that the nvidia.config script is executed. Looking at that file it seems like there is 1) no need for that and 2) it potentially leads to incorrect information (e.g. the target description is not applicable) - if not now, then potentially later e.g. when passes are configured in that file. 
Why not have a config file for nvqc (updating the name of the file or the auto-discovery of targets if needed) or entirely getting rid of the config file and not setting this here (adjusting the condition in line 624)?

Reply:
> Why is this set to nvidia here, just to be set to nvqc further below? It looks like the only consequence this has is that the nvidia.config script is executed.

Yes, that is intentional. We want the `nvidia.config` (or the config file for whatever target is specified by the user) to be executed in order to set the proper shell variables in the `nvq++` driver script.

> it potentially leads to incorrect information (e.g. the target description is not applicable) - if not now, then potentially later e.g. when passes are configured in that file.

I'm not sure I follow this. The `TARGET_DESCRIPTION` variable would be correct for the backend that NVQC is invoking. I.e., if the user selected `tensornet`, then the `TARGET_DESCRIPTION` would be in accordance with that. If passes are added to this config file, we will need to decide which passes we want to run locally vs which ones should run on NVQC, so I suspect additional work would be needed at that point.

> Why not have a config file for nvqc (updating the name of the file or the auto-discovery of targets if needed) or entirely getting rid of the config file and not setting this here (adjusting the condition in line 624)?

I did get rid of the NVQC config file elsewhere in this PR, but as mentioned above, I think this is needed to handle the case when the user is running on a host without GPUs and they run `nvq++ --nvqc <path/to/file.cpp>`.